ReactAndroid/src/main/java/com/facebook/react/views/text/CustomLineHeightSpan.java

```diff
@@ -45,10 +52,13 @@ public void chooseHeight(
       // Show all ascent, descent, bottom, as much top as possible
       fm.top = fm.bottom - mHeight;
     } else {
-      // Show proportionally additional ascent and top
+      // Show proportionally additional ascent / top & descent / bottom
       final int additional = mHeight - (-fm.top + fm.bottom);
-      fm.top -= additional;
-      fm.ascent -= additional;
+
+      fm.top -= additional / 2;
+      fm.ascent -= additional / 2;
+      fm.descent += additional / 2;
+      fm.bottom += additional / 2;
```

Comment:
Why are you using integers? This will cause rounding errors since aditional is an int and 2 is an int so anything that is not divisible by two will be rounded down. (See: https://github.com/facebook/react-native/issues/10712#issuecomment-359028784 )

Reply:
Hi, for us it has solved the problems in the design completely, but I see that there is still something wrong for your design case. Maybe you can open up a new issue for this and see if you can get the attention of other developers, linking to these issues, to get your specific case fixed. Even better to open a new issue, with a PR that fixes it.